### DISCUSSION TITLE
There is no test report when iOS test finished

### DISCUSSION THREAD

**Discussion by Wreos** [leancodepl/patrol#1559] [Q&A] [answered]
Hey, Guys! Could you help me? I have the following error when iOS test is finished. App is starting, test steps are passing, but there is a problem when test is finished - there is no test report like "TEST PASSED". Parallel execution is disabled

```
• Building app with entrypoint test_bundle.dart for iOS simulator (debug)...
	2023-07-20 11:24:42.026 xcodebuild[39691:347320] DVTCoreDeviceEnabledState: DVTCoreDeviceEnabledState_Disabled set via user default (DVTEnableCoreDevice=disabled)
✓ Completed building app with entrypoint test_bundle.dart for iOS simulator (249.0s)
• Running app with entrypoint test_bundle.dart for iOS simulator on simulator iPhone 14 Pro Max...
	2023-07-20 12:13:40.970 xcodebuild[82789:517175] [MT] IDETestOperationsObserverDebug: 18.206 elapsed -- Testing started completed.
	2023-07-20 12:13:40.970 xcodebuild[82789:517175] [MT] IDETestOperationsObserverDebug: 0.000 sec, +0.000 sec -- start
	2023-07-20 12:13:40.970 xcodebuild[82789:517175] [MT] IDETestOperationsObserverDebug: 18.206 sec, +18.206 sec -- end
IDEResultKit: Warning: While writing a result bundle to /Users/*/Library/Developer/Xcode/DerivedData/Runner-eslijftewrqxkqcpzmrwlivtieex/Logs/Test/Test-Runner-2023.07.20_11-27-27-+0200.xcresult, not all contents have been imported, as the Staging directory still contains the following files/directories: ["1_Test]
✓ Completed executing app with entrypoint test_bundle.dart for iOS simulator on simulator iPhone 14 Pro Max (25.8s)

```

<details><summary>Patrol doctor</summary>
<p>
Program adb found in /opt/homebrew/bin/adb
Env var $ANDROID_HOME is not set
Program xcodebuild found in /usr/bin/xcodebuild
Program ideviceinstaller found in /opt/homebrew/bin/ideviceinstaller
</p>
</details> 

<details><summary>Flutter doctor</summary>
<p>
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.10.5, on macOS 13.4.1 22F82 darwin-arm64, locale
    en-DE)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.2)
[✓] VS Code (version 1.80.1)
[✓] Connected device (4 available)
[✓] Network resources
• No issues found!
</p>
</details> 

## Screenshots
![image](https://github.com/leancodepl/patrol/assets/15965366/8d506dc0-2786-40b1-a3ab-21bf4df500db)
![image](https://github.com/leancodepl/patrol/assets/15965366/410e0da7-91f8-4f88-9ac0-517a9b2000fa)
![image](https://github.com/leancodepl/patrol/assets/15965366/3d2c1b00-9f89-4ec4-bd2e-ee3a60ad82a7)
![Screenshot 2023-07-20 at 11 47 50](https://github.com/leancodepl/patrol/assets/15965366/d8b045ac-1a92-4c9f-bbaf-af1ecfd88f8a)
/leancodepl/patrol/assets/15965366/597e6cac-49a2-4472-88ee-5e6859b3f909)
![image](https://github.com/leancodepl/patrol/assets/15965366/394958f1-92e3-4758-a000-22f9cefad422)
![image](https://github.com/leancodepl/patrol/assets/15965366/a8d5917d-bcaf-4e53-89d4-5a0f2828001d)

**Comment by fylyppo** [comment, marked as answer]
Hi, unfortunately this is how it looks like now, you'll be notified about failed tests only.
But if you run with `--verbose` flag you'll get more complete info from run with:
```
Test Case '-[RunnerUITests exampleTest]' passed (7.878 seconds).
        Test Suite 'RunnerUITests' passed at 2023-07-20 22:15:35.030.
                 Executed 1 test, with 0 failures (0 unexpected) in 7.878 (7.879) seconds
        Test Suite 'RunnerUITests.xctest' passed at 2023-07-20 22:15:35.031.
                 Executed 1 test, with 0 failures (0 unexpected) in 7.878 (7.880) seconds
        Test Suite 'All tests' passed at 2023-07-20 22:15:35.031.
                 Executed 1 test, with 0 failures (0 unexpected) in 7.878 (7.882) seconds
```

**Reply by bartekpacia** [reply]
I think the `xcresult` bundle is still generated, but `xcodebuild` (which `patrol_cli` uses internally) just doesn't display it.

But when the test fail, that path is printed, and you can open that `xcresult` in Xcode to get nice view into the results.

**Comment by Wreos** [comment]
Thanks, Guys! I found a reason of problem, it happens because Patrol expects to see something like

`.xctestrun matching qa_iphonesimulator16.4*.xctestrun`

But somehow flavor prefix is being duplicated when I run `patrol test --flavor qa --target integration_test/main_test.dart --verbose
`
`qa_qa_iphonesimulator16.4-arm64-x86_64.xctestrun`

**Generated files**
```Products % ls
Debug-qa-iphonesimulator
qa_qa_iphonesimulator16.4-arm64-x86_64.xctestrun
```

**As result, patrol is not able to find xctestrun file**

```
Looking for .xctestrun matching qa_iphonesimulator16.4*.xctestrun at /Users/aleksandrlozhkovoi/Development/CustomerApp/build/ios_integ/Build/Products
Error: No .xctestrun file was found at /Users/aleksandrlozhkovoi/Development/CustomerApp/build/ios_integ/Build/Products
```

**Comment by Wreos** [comment]
@fylyppo Hey! 😄 Could you help me with on which side it checks the format of name xctestrun file. I have the following problem with duplication of prefix of xctestrun file

> Thanks, Guys! I found a reason of problem, it happens because Patrol expects to see something like
> 
> `.xctestrun matching qa_iphonesimulator16.4*.xctestrun`
> 
> But somehow flavor prefix is being duplicated when I run `patrol test --flavor qa --target integration_test/main_test.dart --verbose ` `qa_qa_iphonesimulator16.4-arm64-x86_64.xctestrun`
> 
> **Generated files**
> 
> ```
> Debug-qa-iphonesimulator
> qa_qa_iphonesimulator16.4-arm64-x86_64.xctestrun
> ```
> 
> **As result, patrol is not able to find xctestrun file**
> 
> ```
> Looking for .xctestrun matching qa_iphonesimulator16.4*.xctestrun at /Users/aleksandrlozhkovoi/Development/CustomerApp/build/ios_integ/Build/Products
> Error: No .xctestrun file was found at /Users/aleksandrlozhkovoi/Development/CustomerApp/build/ios_integ/Build/Products
> ```
